subworkflows/local/profiling.nf

```diff
@@ -362,24 +362,28 @@ workflow PROFILING {
         ch_input_for_krakenuniq =  ch_input_for_profiling.krakenuniq
             .map {
                 meta, reads, db_meta, db ->
-                    [[id: db_meta.db_name, single_end: meta.single_end], reads, db_meta, db]
+                    def seqtype = reads[0].name.matches(".*a.gz\$|.*a\$") ? 'fasta' : 'fastq'
```

Comment:
For clarity, I would make the first part non-greedy and contain at least one character (`.+?`). Instead of an OR, I would make the last part optional (`(\.gz)?`). Admittedly, this is off the top of my head but tested on https://regex101.com/. Could also check for an `f` in the extension. And I like to make the regex a bit groovier so backslash escaping is not needed.

```suggestion
                    def seqtype = (reads[0].name ==~ /.+?\.f\w{0,3}a(\.gz)?$/) ? 'fasta' : 'fastq'
```